File: Python/refer-offer/qing_wa_tiao_tai_jie_wen_ti_lcof.py
def numWays(n: int) -> int:
    if n == 0 or n == 1:
        return 1
    if n == 2:
        return 2
    f1, f2 = 1, 2
    for i in range(n - 2):
        temp = (f1 + f2) % 1000000007
        f1 = f2
        f2 = temp
    
    return f2

if __name__ == "__main__":
    print(numWays(2))

File: src/tmp/Java/src/com/dbc/algorithm/ReferOffer/QingWaTiaoTaiJieWenTiLcof.java
package com.dbc.algorithm.ReferOffer;

public class QingWaTiaoTaiJieWenTiLcof {
    public int numWays(int n) {
        if (n == 0 || n == 1) return 1;
        if (n == 2) return 2;
        int f1 = 1, f2 = 2;
        for (int i = 0; i < n - 2; i++) {
            int temp = (f1 + f2) % 1000000007;
            f1 = f2;
            f2 = temp;
        }
        return f2;
    }
}
